### PR TITLE
feat(Guild): allow description and features in edit

### PR DESCRIPTION
**Pull request by almostSouji**
**Please describe the changes this PR makes and why it should be merged:**

- `Guild#edit` now allows modification of features (`COMMUNITY`, to enable community features)
- `Guild#edit` now allows the discovery description to be modified

as per https://github.com/discord/discord-api-docs/pull/1764

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/structures/Guild.js [hunk 1]
```diff
@@ -926,6 +926,7 @@ class Guild extends Base {
 
   /**
    * The data for editing a guild.
+   * <info>Only the `COMMUNITY` feature can be modified</info>
```
Comment (vladfrangu) [inline src/structures/Guild.js:929]: Do we want to document this? It'd mean we have to update it every time Discord adds new ones (also pretty sure it's NOT the only one)

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/structures/Guild.js (hunk 1 above)
Comment (almostSouji) [reply src/structures/Guild.js:929]: every other feature is a privilege you can't self-grant or disable so far

File: typings/index.d.ts [hunk 2]
```diff
@@ -2786,6 +2786,8 @@ declare module 'discord.js' {
     rulesChannel?: ChannelResolvable;
     publicUpdatesChannel?: ChannelResolvable;
     preferredLocale?: string;
+    description?: string;
```
Comment (kyranet) [inline typings/index.d.ts:2789]: Those properties should be sorted alphabetically

**Review by kyranet — CHANGES_REQUESTED** [review]

**Comment by SpaceEEC** [comment]
I'm unable to set a description (just silently fails on Discord's end), probably due to not being on discovery.

Further enabling community only works when the payload also includes a rules and public updates channel (does not matter if you previously set them).
Otherwise you are getting the very helpful and totally not misleading error `This feature has been temporarily disabled` (The code is `40006`, if somebody is interested).

I don't think there is much we can do about either though.

**Review by SpaceEEC — APPROVED** [review]

File: typings/index.d.ts (hunk 2 above)
Comment (vladfrangu) [inline typings/index.d.ts:2789]: Can't you also set descriptions to null?

Comment (almostSouji) [reply typings/index.d.ts]: it does! fixed

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by catcosmicice — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**